modAionImpl/src/org/aion/zero/impl/db/RecoveryUtils.java

```diff
@@ -93,6 +94,9 @@ public static void pruneAndCorrect() {
         store.pruneAndCorrect();
         store.flush();
 
+        // compact database after the changes were applied
+        blockchain.getRepository().compact();
+
```

Comment:
Will the compaction take long after too many update?

Reply:
I tested it for 300k blocks and it was minor compared to the total pruning time.